meeting-notes/2022-05-14.md

```diff
@@ -0,0 +1,261 @@
+# 5/14/22
+
+@pcohen, @rntz
+
+## TODOs
+
+- [ ] Phil JSON:
+  - add cursor selection ranges to JSON output
+- [ ] rntz JSON:
+  - Add selections (including the anchor)
+  - Multiple cursors
+  - Buffer filename should be null if nonexistent
+  - camelCase
+
+@rntz sez: "maybe we should put all these files in a ~/.cursorless/ directory?"
+
+## Notes
+
+**what phil's fork of the cursorless vscode plugin does**
+
+1. writes hats to a file when there's a change to those hats
+
+**phil next steps**
+
+- Migrate "watch JB/Emacs editor state and tell VS Code to navigate to that same
+  location (thus updating the hats)"logic to cursorless fork
+  (file watcher)
+
+- Make new JetBrains plugin also publish standardized \~/.editor-state.json,
+  since it\'s current JSON is overly complicated for this use case
+
+- Export forked changes in knausj VSCode command server
+
+- Export changes from custom command server
+
+**rntz next steps**
+
+- ~~Install and get building (./build)~~
+
+- Get emacs reading \~/.vscode-hats.json and applying hats
+
+- Think about how commands will be sent to visual studio code\
+  *issue: *vscode "command server"waits for a keystroke to tell it to read a
+  file talon writes to execute cursorless commands. just use a file watcher?
+
+    - Get the knausj VSCode\* actions (most importantly
+      vscode_with_plugin_and_wait) to work without VS Code being focused (ie
+      from the talon REPL)
+
+*hooks I might need:*
+
+https://www.gnu.org/software/emacs/manual/html_node/elisp/Change-Hooks.html
+
+- buffer contents change: after-change-functions
+
+- view changes
+
+- current buffer changes
+
+- cursor/selection change
+
+post-command-hook?
+
+**both next steps**
+
+- Think about temporary files for contents
+
+**things to standardize:**
+
+- paths of JSON files
+
+    - \~/.editor-state.json
+
+    - \~/.vscode-hats.json
+
+- schema of JSON files
+
+**Old (before 5/14/22)**
+
+- serialize editor state to file, at the moment:
+
+- a serial number
+
+- current file path
+
+- top & bottom visible lines
+
+edge case: long lines without wrapping, does cursorless hat them?
+
+looks like it does.
+
+- where the cursors/selections are
+
+- watch file from talon or vscode
+
+- vscode draws hats and serializes to a file
+
+- editor watches hat-file and draws hats
+
+problem: how to draw hats in emacs? what if hats are out of date wrt file?
+
+- issue a cursorless command
+
+- goes to vscode via a command server
+
+- vscode edits the file and saves, tells talon it\'s done
+
+- editor either watches the file or is told to reload
+
+eg. "air bat cap <cursorless command> drum each fine"
+
+**Design Stuff**
+
+**Things to think about**
+
+- Think about how cursorless-talon gets published, because it\'s part of
```

Comment:
I have recorded a video about how this release process works; hopefully will release in the next couple weeks

Reply:
+1. This is mostly about our fork version of `cursorless-talon`, because there might be changes we might need to make there.

If we will have changes to it long term, we'll need to set up the same subrepo exporting system. That way we can make the changes in https://github.com/phillco/cursorless (the one source of truth), but people could still clone https://github.com/phillco/cursorless-talon into their user directory.